biotrainer/config/configurator.py

```diff
@@ -317,7 +392,273 @@ def _get_config_maps(
+    ):
+        """
+        Logs existing sequence and label files based on the protocol.
+
+        Args:
+            protocol (Protocol): The selected protocol.
+            config_map (Dict[str, ConfigOption]): Mapping of configuration option names to their instances.
+        """
+        if protocol in Protocol.per_sequence_protocols():
+            sequence_file_option = config_map.get("sequence_file")
+            if sequence_file_option and os.path.exists(sequence_file_option.value):
+                logger.info("sequence_file already exists. Overwriting it.")
+
+        if protocol in Protocol.per_residue_protocols():
+            sequence_file_option = config_map.get("sequence_file")
+            labels_file_option = config_map.get("labels_file")
+
+            if sequence_file_option and os.path.exists(sequence_file_option.value):
+                logger.info("sequence_file already exists. Overwriting it.")
+            if labels_file_option and os.path.exists(labels_file_option.value):
+                logger.info("labels_file already exists. Overwriting it.")
+
+    def _load_and_split_dataset(self, hf_map: Dict[str, ConfigOption]) -> Tuple[List[str], List[Any], List[str]]:
+        """
+        Loads the HuggingFace dataset and splits it into sequences, targets, and set_values.
+
+        Args:
+            hf_map (Dict[str, ConfigOption]): Mapping of hf_dataset option names to their instances.
+
+        Returns:
+            Tuple[List[str], List[Any], List[str]]: Sequences, targets, and corresponding set values.
+
+        Raises:
+            ConfigurationException: If required columns are missing or dataset loading fails.
+        """
+        path = hf_map["path"].value
+        sequence_column = hf_map["sequence_column"].value
+        target_column = hf_map["target_column"].value
+        subset_name = hf_map["subset"].value if "subset" in hf_map else None
+
+        logger.info(f"Loading HuggingFace dataset from path: {path}")
+
+        if not subset_name:
+            try:
+                dataset = load_dataset(path)
+            except ValueError as e:
+                raise ConfigurationException(
+                    "You should specify the 'subset' option in the config file.\n"
+                    f"Error: {e}"
+                )
+        else:
+            try:
+                dataset = load_dataset(path, subset_name)
+            except ValueError as e:
+                raise ConfigurationException(
+                    "You should select a 'subset' from the Available list.\n"
+                    "If the Available 'subset' is 'default', you can remove the 'subset' option from the config file.\n"
+                    f"Error: {e}"
+                )
+
+        # Collect all available splits
+        available_splits = list(dataset.keys())
+        if not available_splits:
+            raise ConfigurationException(
+                f"No splits found in the dataset at path '{path}'."
```

Comment:
I'd suggest to move all of this logic to the FASTA utility, because it really is concerned about converting the huggingface dataset object to fasta files, not so much about the configuration options themselves.